### PR TITLE
chore: experiment config slots to comply with constraint max slots

### PR DESCRIPTION
**Pull request by amandavialva01**
## Ticket

[CM-569]

## Description
Experiment config slots to comply with constraint max slots. 

## Test Plan
CI Passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[CM-569]: https://hpe-aiatscale.atlassian.net/browse/CM-569?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/configpolicy/utils.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func ValidateExperimentConfig(
 		if cp.InvariantConfig.RawResources != nil {
 			checkAgainstGlobalPriority(priorityEnabledErr, cp.InvariantConfig.RawResources.RawPriority)
 			if err := checkConstraintConflicts(cp.Constraints, cp.InvariantConfig.RawResources.RawMaxSlots,
-				cp.InvariantConfig.RawResources.RawSlotsPerTrial, cp.InvariantConfig.RawResources.RawPriority); err != nil {
+				cp.InvariantConfig.RawResources.RawSlots, cp.InvariantConfig.RawResources.RawPriority); err != nil {
```
Comment (stoksc) [inline master/internal/configpolicy/utils.go:97]: I don't understand this change. Experiments use slots per trial, not slots, right?

Comment (kkunapuli) [reply master/internal/configpolicy/utils.go:97]: If that's the case, @amandavialva01 can you also update the logic [here](https://github.com/determined-ai/determined/blob/9efd96df058d33c4e7524e486c0629aea1283aa5/master/internal/configpolicy/task_config_policy.go#L91) to use slots per trial?

Comment (amandavialva01) [reply master/internal/configpolicy/utils.go:97]: _Ooh_ my mistake! Changed this. I think the actual discrepancy was coming from the < / > issue that Kristine mentioned

Comment (kkunapuli) [reply master/internal/configpolicy/utils.go]: Synced with Bradley - for NTSC we'll compare "slots" to "constraints.resources.max_slots" and for experiments we'll compare "slotsPerTrial" to "constraints.resources.max_slots".

Comment (amandavialva01) [reply master/internal/configpolicy/utils.go]: Makes sense! Yea I just saw the comment in the `ResourcesConfig` struct, somehow glossed over that before lol

File: master/internal/configpolicy/utils_test.go [hunk 2]
```diff
@@ -309,3 +310,45 @@ func TestUnmarshalJSONNTSC(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckConstraintsConflicts(t *testing.T) {
```
Comment (kkunapuli) [inline master/internal/configpolicy/utils_test.go:314]: Nice tests!

**Pull request merged**